lib/msal-browser/src/crypto/PkceGenerator.ts

```diff
@@ -25,6 +25,47 @@ const RANDOM_BYTE_ARR_LENGTH = 32;
  * This file defines APIs to generate PKCE codes and code verifiers.
  */
 
+let pkceCode: PkceCodes | undefined;
+
+/**
+ * Pre-generates PKCE codes and stores it in local variable
+ * @param performanceClient
+ * @param logger
+ * @param correlationId
+ */
+export async function preGeneratePkceCodes(
+    performanceClient: IPerformanceClient,
+    logger: Logger,
+    correlationId: string
+): Promise<void> {
+    logger.verbose("Generating new PKCE codes");
+    pkceCode = await generatePkceCodes(
+        performanceClient,
+        logger,
+        correlationId
+    );
+    return Promise.resolve();
+}
+
+/**
+ * Provides pre-generated PKCE codes, if any
+ * @param logger
+ */
+export function getPreGeneratedPkceCodes(
+    performanceClient: IPerformanceClient,
+    logger: Logger,
+    correlationId: string
+): PkceCodes | undefined {
+    logger.verbose("Attempting to pick up pre-generated PKCE codes");
+    const res = pkceCode;
```

Comment:
Objects are passed by reference. When you reassign pkceCode to undefined below you may unexpectedly receive undefined when referencing res as well.

```suggestion
    const res = {...pkceCode};
```

Reply:
Good point, fixed